test/unittest/dds/xtypes/type_representation/TypeObjectUtilsTests.cpp

```diff
@@ -461,93 +495,398 @@ TEST(TypeObjectUtilsTests, build_plain_map_s_type_defn_inconsistencies)
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, minimal_indirect_id, flags, minimal_alias_id), InvalidArgumentError);
+
+    EXPECT_NO_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, minimal_alias_id, flags, key_identifier));
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, minimal_alias_id, flags, complete_typeid), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, minimal_alias_id, flags, complete_indirect_id), InvalidArgumentError);
+    EXPECT_NO_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, minimal_alias_id, flags, complete_alias_id));
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, minimal_alias_id, flags, minimal_typeid), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, minimal_alias_id, flags, minimal_indirect_id), InvalidArgumentError);
+    EXPECT_NO_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, minimal_alias_id, flags, minimal_alias_id));
+
+
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
                 complete_header, 10, test_identifier, flags, key_identifier), InvalidArgumentError);
-    // TypeIdentifier consistent with minimal header
-    EXPECT_NO_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
-                minimal_header, 10, test_identifier, flags, key_identifier));
-    // TypeIdentifier inconsistent with fully-descriptive header
-    EXPECT_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
-                fully_descriptive_header, 10, test_identifier, flags, key_identifier), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                complete_header, 10, test_identifier, flags, complete_typeid), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                complete_header, 10, test_identifier, flags, complete_indirect_id), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                complete_header, 10, test_identifier, flags, complete_alias_id), InvalidArgumentError);
```

Comment:
This case is correct and should not throw. In fact, using an alias of a primitive type should use a non EK_BOTH header. The equivalence kind should be according to the alias TypeIdentifier.

Reply:
Changed aliases behabiour in [d3245e1](https://github.com/eProsima/Fast-DDS/pull/4496/commits/d3245e16dd687c62ce397be8018c718828d2bc6e)